packages/client/src/objectSet/ObjectSet.ts

```diff
@@ -15,41 +15,44 @@
  */
 
 import type {
+  InterfaceNamesFrom,
+  InterfacePropertyKeysFrom,
   ObjectInfoFrom,
+  ObjectPropertyKeysFrom,
   ObjectTypesFrom,
   OntologyDefinition,
-  PropertyKeysFrom,
 } from "@osdk/api";
 import type { FetchPageOrThrowArgs } from "../object/fetchPageOrThrow";
-import type { OsdkObjectFrom } from "../OsdkObjectFrom";
+import type { OsdkInterfaceFrom, OsdkObjectFrom } from "../OsdkObjectFrom";
 import type { PageResult } from "../PageResult";
 import type { AggregationsResults, WhereClause } from "../query";
 import type { AggregateOpts } from "../query/aggregations/AggregateOpts";
 import type { LinkTypesFrom } from "./LinkTypesFrom";
 
 export type ObjectSet<
   O extends OntologyDefinition<string>,
-  K extends ObjectTypesFrom<O>,
-> = BaseObjectSet<O, K>; // & SearchAround<O, K>;
-
-// GOTTA DO THIS STILL
-export type SearchAround<
-  O extends OntologyDefinition<string>,
-  K extends ObjectTypesFrom<O>,
-> = {
-  [L in LinkTypesFrom<O, K> & string as `searchAround_${L}`]: () => ObjectSet<
-    O,
-    L
-  >; // TODO accept args?
-};
+  K extends ObjectTypesFrom<O> | InterfaceNamesFrom<O>,
+> = BaseObjectSet<O, K>;
 
 export interface BaseObjectSet<
   O extends OntologyDefinition<any>,
-  K extends ObjectTypesFrom<O>,
+  K extends ObjectTypesFrom<O> | InterfaceNamesFrom<O>,
```

Comment:
Interfaces and Objects are guaranteed to have unique names within the union right?